harness/determined/exec/launch_autohorovod.py

```diff
@@ -0,0 +1,156 @@
+"""
+launch_autohorovod.py is the default launch layer for Determined.
+
+It launches the entrypoint script under horovodrun when slots_per_trial>1, or as a regular
+subprocess otherwise.
+"""
+
+import distutils.util
+import json
+import logging
+import os
+import sys
+
+import simplejson
+
+import determined as det
+import determined.common
+from determined import gpu, horovod, layers
+from determined.common import constants, storage
+from determined.common.api import certs
+
+ENVIRONMENT_VARIABLE_KEYS = {
+    "DET_MASTER_ADDR",
+    "DET_MASTER_PORT",
+    "DET_AGENT_ID",
+    "DET_SLOT_IDS",
+    "DET_CONTAINER_ID",
+    "DET_USE_GPU",
+    "DET_EXPERIMENT_ID",
+    "DET_TRIAL_ID",
+    "DET_TRIAL_SEED",
+    "DET_EXPERIMENT_CONFIG",
+    "DET_HPARAMS",
+    "DET_LATEST_CHECKPOINT",
+    "DET_LATEST_BATCH",
+    "DET_RENDEZVOUS_PORT",
+    "DET_TRIAL_RUNNER_NETWORK_INTERFACE",
+    "DET_TASK_TOKEN",
+    "DET_TRIAL_RUN_ID",
+    "DET_TASK_ID",
+    "DET_RENDEZVOUS_INFO",
+}
+
+
+def main() -> int:
+    for k in ENVIRONMENT_VARIABLE_KEYS:
```

Comment:
we could use sets here. like set(ENVIRONMENT_VARIABLE_KEYS) - set(os.environ) = missing vars